src/awkward/contents/unionarray.py

```diff
@@ -1583,7 +1583,7 @@ def _to_backend(self, backend: Backend) -> Self:
 
     def _is_equal_to(self, other, index_dtype, numpyarray):
         return (
-            self.tags == other.tags
+            self.tags.is_equal_to(other.tags)
```

Comment:
Yikes! That's right, this `==` is either not defined on Index or it would be non-coercible to a scalar Boolean.